File: dart/bubble_sort.dart
import 'dart:math';

void bubbleSort(List<int> arr) {
  int n = arr.length;
  for (int i = 0; i < n - 1; i++) {
    for (int j = i + 1; j < n; j++) {
      if (arr[i] > arr[j]) {
        int temp = arr[j];
        arr[j] = arr[i];
        arr[i] = temp;
      }
    }
  }
}

List<int> generateArray(int n) {
  final random = Random(42); // Fixed seed for reproducible results
  return List.generate(n, (index) => random.nextInt(10000));
}

void main(List<String> args) {
  if (args.isEmpty) {
    print('Usage: bubble_sort <n>');
    return;
  }
  
  final n = int.parse(args[0]);
  final arr = generateArray(n);
  
  final stopwatch = Stopwatch()..start();
  bubbleSort(arr);
  stopwatch.stop();
  
  print('Dart: bubble_sort($n) = sorted');
  print('Time: ${stopwatch.elapsedMilliseconds}ms');
}

File: rust/bubble_sort.rs
use std::env;
use std::time::Instant;

fn bubble_sort(arr: &mut [i32]) {
    let n = arr.len();
    for i in 0..n - 1 {
        for j in 0..n - i - 1 {
            if arr[j] > arr[j + 1] {
                arr.swap(j, j + 1);
            }
        }
    }
}

fn generate_array(n: usize) -> Vec<i32> {
    // Simple seeded random for reproducible results
    let mut seed: u32 = 42;
    let mut arr = Vec::with_capacity(n);
    
    for _ in 0..n {
        seed = seed.wrapping_mul(1103515245).wrapping_add(12345);
        arr.push((seed % 10000) as i32);
    }
    
    arr
}

fn main() {
    let args: Vec<String> = env::args().collect();
    if args.len() < 2 {
        println!("Usage: bubble_sort <n>");
        return;
    }
    
    let n: usize = args[1].parse().unwrap();
    let mut arr = generate_array(n);
    
    let start = Instant::now();
    bubble_sort(&mut arr);
    let duration = start.elapsed();
    
    let time_ms = duration.as_millis();
    
    println!("Rust: bubble_sort({}) = sorted", n);
    println!("Time: {}ms", time_ms);
}